### ISSUE TITLE
Consider adding Leave Studio button

### ISSUE THREAD

**Issue by apple502j** [LLK/scratch-www#5400]
With the current migration of studios, it's a good time to implement new features.

## Why add new feature, now?
Users expect that big changes come with some good things (for them; not like "finally free from Django stuff") - and given that Scratch Team rarely adds new features once released, this is a good opportunity to implement this.

## Why studio leave button?
First, **user's convenience** - in big studios, it's very hard to see the username in crowded area. It is also possible that the server/the user's computer cannot handle items such deep in the list. This makes it impossible to leave studios once joined.

Second, **user's security** - some users do want to leave studios, so they usually run userscripts or browser extensions which include such features (that are advertised against the guidelines, or outside the Scratch Team's moderation). This obviously has negative impact to the overall security of the system, because an important feature is implemented by the community in ways that, when they go wrong, can lead to account compromise.

Third, **the reputation** - implementing long-requested feature is a good way to increase the reputation of the Scratch Team.

## Designs
Studio leave buttons should be placed at the top of the curator **page** (not the curator section, since that doesn't help when there are many curators.) The button should have confirmation dialog just to make sure people do not click by accident.

The button should not be shown for non-curator/managers. It may or may not be shown for owners; however, if shown, it should be disabled and have the tooltip like "Owners cannot leave studios."

## Alternatives
One alternative method I considered is to bump the curator list item so that the user themselves show at the top. However, this does not help when **there are many managers** and you are the curator. This may also involve excessive API calls due to pagination.

Simple "studio leave" button does not have this problem if placed correctly, and should not require additional API calls, as all it needs is "whether the user is curator/manager" and "whether the user owns the studio", both available on Redux.

Suggestion topic on Leave Studio button: https://scratch.mit.edu/discuss/post/908994/
Note that because this suggestion has been suggested so many times, it might not be the original one.

**Comment by WorldLanguages** [comment]
Apparently, my "how to leave a studio without finding yourself" tutorial is one of the 2500 most loved projects on Scratch, with over 3,000 loves and 35,000 views. I've been waiting for years for the moment where I can finally unshare my own tutorial for being unnecessary 👌

**Comment by jemunm378** [comment]
I think it’s a great idea. It also could be below the "delete" button in "my stuff”.

**Comment by Ihatr1** [comment]
There's already console code to do this, all we need is an accessible button.

It's really easy to implement, any fool can do it in a few minutes if they at least know some HTML and CSS

**Comment by apple502j** [comment]
(please be respectful, the rules apply here too!)

**Comment by gosoccerboy5** [comment]
> There's already console code to do this, all we need is an accessible button.
> 
> It's really easy to implement, any fool can do it in a few minutes if they at least know some HTML and CSS

It would be nice if you could make a mockup, proving you're not a fool. :P

**Comment by return-misaPuding** [comment]
I agree. Using userscript and such to leave crowded studio is unsecure and inconvenient. Now that studios are being migrated to scratch-www, it´s the right time to finally implement this

**Comment by locness3** [comment]
No update on this?

**Comment by turtle846** [comment]
i really want this

**Comment by ashishagarwal2023** [comment]
I might look onto it.
